test/upgrade-tests.sh

```diff
@@ -3,7 +3,7 @@
 # shellcheck disable=SC1091,SC1090
 source "$(dirname "${BASH_SOURCE[0]}")/lib.bash"
 
-set -Eeuox pipefail
+set -Eeuo pipefail
```

Comment:
That's too much output. It's hard to know what's going on. Rather, add `logger.info` and `logger.debug` statements to make it more visible.

Reply:
There's never enough output in CI. For local runs - OK - you might need less logging. 
Using debug is not enough either. For example, the timeout function will log something like `logger.debug 'Execution failed: [[ $(oc get knativeserving knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") != True ]]. Waiting 5 sec (5/900)...'` but you don't know what the status actually is. With -x you know it. It also forces everyone to think whether to use INFO or DEBUG. I'd say in CI we always want to see as much output as we can except for cases when the logs exceed some quota etc.
What about enabling -x only if it runs in CI? Is it possible? For local runs it would stay without it

Reply:
It was only for the local runs that I removed it, and even for CI I wouldn't want it, personally.

That being said, most of the noise was due to the logging level logic itself. IMHO, the `-x` option obviates the need for any logging at all, thereby keeping the code base smaller